index.js

```diff
@@ -20,46 +20,54 @@ app.listen(3000, function(req, res){
 });
 
 //search query
-app.get('/searchJobs', function(req, res){
+app.get('/searchJobs', search);
 	
-	let trackId = [];
-	req_id = uuidv4();
-	trackId.push(req_id);
-	console.log(trackId);
-	var filename = req_id + '.json';
-	console.log(filename);
-	var queryParameter = req.query;
-	request({
-    	uri: 'https://jobs.github.com/positions.json?description='+ req.query.description +'&location='+ req.query.location,
-    	
-  	},function(err, res, body){
-  		if(!err && res.statusCode === 200){
-  			let data = [];
-  			console.log(res.statusCode) 
-    		console.log(res.headers['content-type'])
-    		data.push(body);
-    		fs.writeFile(`./files/${filename}`, data, function (err) {
-  				if (err) throw err;
-  				console.log('Saved!');
-			});
+	function search(req, res){
+		let trackId = [];
+		var req_id = uuidv4();
+		module.exports = trackId;
+		trackId.push(req_id);
+		console.log(trackId);
+		var filename = req_id + '.json';
+		console.log(filename);
+		var queryParameter = req.query;
+		request({
+	    	uri: 'https://jobs.github.com/positions.json?description='+ req.query.description +'&location='+ req.query.location,
+	    	
+	  	},function(err, res, body){
+	  		if(!err && res.statusCode === 200){
+	  			let data = [];
+	  			console.log(res.statusCode) 
+	    		console.log(res.headers['content-type'])
+	    		data.push(body);
+	    		fs.writeFile(`./files/${filename}`, data, function (err) {
+	  				if (err) throw err;
```

Comment:
If you `throw` the error I believe it will crash the server. Just print the error that's enough. Also `trackId.push` can be done only after the successful write into the file. Because if the `writeFile` function throws an error then we would be left with no file creation but the respective trackId is been pushed into the array. So let's add it only after successful creation of the file.